android/build.gradle

```diff
@@ -11,7 +18,32 @@ buildscript {
 
 allprojects {
   repositories {
+    mavenLocal()
     jcenter()
-    maven { url "$projectDir/../../node_modules/react-native/android" }
+    maven {
+        // Refer to the local install of react-native in the node_modules folder
+        // react-native should be installed since it's a peer dependency
+        url "$projectDir/../../react-native/android"
+    }
+  }
+}
+
+android {
+  compileSdkVersion 23
+  buildToolsVersion "23.0.3"
+
+  defaultConfig {
+    minSdkVersion 16
+    targetSdkVersion 23
   }
-}
\ No newline at end of file
+
+  lintOptions {
+    disable 'InvalidPackage'
+  }
+}
+
+dependencies {
+  compile 'com.facebook.react:react-native:+'
```

Comment:
This is definitely something that we should all look more into. I have in the past, had trouble with these versioning specificity changes, and have others. It is perhaps an issue with different Gradle versions.